From c09aae648837849138129fc4ad4fc342e14378a5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Barnab=C3=A1s=20P=C5=91cze?=
 <barnabas.pocze@ideasonboard.com>
Date: Thu, 23 Jul 2026 13:35:08 +0200
Subject: [PATCH 33/68] ipa: libipa: agc: Use full range as default
 `FrameDurationLimits`
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Currently the default value for `FrameDurationLimits` is meaningless
because the full range is used by default. So advertise the full range.

Signed-off-by: Barnabás Pőcze <barnabas.pocze@ideasonboard.com>
Reviewed-by: Jacopo Mondi <jacopo.mondi@ideasonboard.com>
Reviewed-by: Stefan Klug <stefan.klug@ideasonboard.com>
(cherry picked from commit 9d2d7b46eab96acda67e1f04c9cf58a8f3b5c6ce)
---
 src/ipa/libipa/agc.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ipa/libipa/agc.cpp b/src/ipa/libipa/agc.cpp
index 0dbf971f8..345f09e9a 100644
--- a/src/ipa/libipa/agc.cpp
+++ b/src/ipa/libipa/agc.cpp
@@ -449,7 +449,7 @@ int AgcAlgorithm::configure(agc::Session &session, agc::ActiveState &state,
 	};
 	config.ctrlMap[&controls::FrameDurationLimits] = ControlInfo{
 		frameDurations[0], frameDurations[1],
-		Span<const int64_t, 2>{ { frameDurations[2], frameDurations[2] } },
+		Span<const int64_t, 2>{ { frameDurations[0], frameDurations[1] } },
 	};
 
 	const auto add = [&](const ControlId &cid, const auto &automatic, const auto &manual) {
